src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.LegacyEditor/RazorSourceSpan.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Razor.ExternalAccess.LegacyEditor;
+
+internal record struct RazorSourceSpan(
+    string? FilePath,
+    int AbsoluteIndex,
+    int LineIndex,
+    int CharacterIndex,
+    int Length,
+    int LineCount,
+    int EndCharacterIndex)
+{
+    public RazorSourceSpan(int absoluteIndex, int length)
+        : this(filePath: null, absoluteIndex, lineIndex: -1, characterIndex: -1, length)
+    {
+    }
+
+    public RazorSourceSpan(string? filePath, int absoluteIndex, int lineIndex, int characterIndex, int length)
+        : this(filePath, absoluteIndex, lineIndex, characterIndex, length, LineCount: 0, EndCharacterIndex: 0)
+    {
+    }
+
+    public RazorSourceSpan(int absoluteIndex, int lineIndex, int characterIndex, int length)
+        : this(filePath: null, absoluteIndex, lineIndex, characterIndex, length)
+    {
+    }
+
+    public readonly bool Equals(RazorSourceSpan other)
```

Comment:
Why does equality not consider `LineCount` or `EndCharacterIndex`? Is is simply that they're effectively calculated properties? If so, does it make sense to just make them calculated properties?

Reply:
Sadly, the reason is that the compiler's `SourceSpan` leaves these out of equality for some reason. I have no idea why that is, but I need to match the equality or tests fail.

Reply:
Gotcha. Probably worth a comment.